plugins/modules/smtp.py

```diff
@@ -0,0 +1,301 @@
+        module.warn("api_entry: " + str(api_entry))
+        module.warn("change_needed: " + str(change_needed))
+
+    return module_entry, change_needed
+
+
+def modify_smtp_config(
+    module: AnsibleModule, rest_client: RestClient
+) -> Tuple[bool, dict, dict]:
+    # GET method to get the SMTP config by UUID
+    smtp = SMTP.get_by_uuid(module.params, rest_client)
+
+    # If SMTP config doesn't exist, raise an exception (error)
+    if not smtp:
+        raise errors.ScaleComputingError("SMTP: There is no SMTP configuration.")
+
+    # Otherwise, continue with modifying the configuration
+    before = smtp.to_ansible()
+    old_state = smtp.get_state(
+        rest_client
+    )  # get the state of SMTP config before modification
+
+    new_smtp_server, new_smtp_server_change_needed = build_entry(
+        before.get("smtp_server"), module.params["server"]
+    )
+    new_port, new_port_change_needed = build_entry(
+        before.get("port"), module.params["port"]
+    )
+    new_use_ssl, new_use_ssl_change_needed = build_entry(
+        before.get("use_ssl"), module.params["use_ssl"]
+    )
+    new_use_auth, new_use_auth_change_needed = build_entry(
+        before.get("use_auth"), module.params["requires_auth"]
+    )
+    new_auth_user, new_auth_user_change_needed = build_entry(
+        before.get("auth_user"), module.params["username"]
+    )
+    new_auth_password, new_auth_password_change_needed = build_entry(
+        before.get("auth_password"), module.params["password"]
+    )
+    new_from_address, new_from_address_change_needed = build_entry(
+        before.get("from_address"), module.params["from_address"]
+    )
+
+    # Init return values and return if no changes were made
+    change, record, diff = (
+        new_smtp_server_change_needed
+        or new_port_change_needed
+        or new_use_ssl_change_needed
+        or new_use_auth_change_needed
+        or new_auth_user_change_needed
+        or new_auth_password_change_needed
+        or new_from_address_change_needed,
+        old_state,
+        dict(before=old_state, after=old_state),
+    )
+    module.log("change: " + str(change))
+    if not change:
+        return change, record, diff
+
+    # Set payload
+    payload = dict(
+        smtpServer=new_smtp_server,
+        port=new_port,
+        useSSL=new_use_ssl,
+        useAuth=new_use_auth,
+        authUser=new_auth_user,
+        authPassword=new_auth_password,
+        fromAddress=new_from_address,
+    )
+
+    # Set the task tag
+    # update_record -> PATCH
+    update_task_tag = rest_client.update_record(
+        endpoint="{0}/{1}".format("/rest/v1/AlertSMTPConfig", smtp.uuid),
+        payload=payload,
+        check_mode=module.check_mode,
+    )
+
+    TaskTag.wait_task(rest_client, update_task_tag)  # wait for task to finish
+
+    record = SMTP.get_state(rest_client)
+    diff = dict(before=old_state, after=record)
+    change = old_state != record
```

Comment:
Here you compute `change` a second time. Why? You can also assert that `old_state != record` is True.